src/material-experimental/mdc-list/list-item-sections.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directive, ElementRef, Inject, Optional} from '@angular/core';
+import {LIST_OPTION, ListOption} from './list-option-types';
+
+/**
+ * Directive capturing the title of a list item. A list item usually consists of a
+ * title and optional secondary or tertiary lines.
+ *
+ * Text content for the title never wraps. There can only be a single title per list item.
+ */
+@Directive({
+  selector: '[matListItemTitle]',
+  host: {'class': 'mat-mdc-list-item-title mdc-list-item__primary-text'},
+})
+export class MatListItemTitle {
+  constructor(public _elementRef: ElementRef<HTMLElement>) {}
+}
+
+/**
+ * Directive capturing a line in a list item. A list item usually consists of a
+ * title and optional secondary or tertiary lines.
+ *
+ * Text content inside a line never wraps. There can be at maximum two lines per list item.
+ */
+@Directive({
+  selector: '[matListItemLine]',
```

Comment:
Backwards compatibility is unfortunately a little more complicated due to need of having an explicit title if a line is used. 

Given that this is not a simple drop-in-replacement anyway, I think we should just get rid of it as we'll need some rather advanced migration anyway. Thoughts?

Reply:
Ah, I was hoping that supporting `mat-line` would make it so users could migrate without needing any changes to their template (of course they might decide to tweak it after to get it looking exactly how they want it). But it sounds like that's not really possible so in that case I agree with just dropping it.

I still want the migration script to be able to migrate existing usages to something functional at least, even if it does have to modify the template. If our migration script can understand the DOM hierarchy maybe it can convert the first mat-line to a title and the rest to lines or something like that

Reply:
yeah, that is what I had in mind as well. We can talk about the details offline. I think I had a small note about the migration in the design doc as well